BGL/include/CGAL/boost/graph/named_params_helper.h

```diff
@@ -112,9 +113,196 @@ namespace CGAL {
     typedef typename boost::property_traits<PMap>::value_type type;
   };
 
+  namespace internal_np
+  {
+  template<typename key_value, class PMap, class Graph, typename Tag>
+  struct Index_map_initializer{
+    void operator()(PMap, const Graph& )
+    {}
+  };
+
+  template< class PMap, class Graph>
+  struct Index_map_initializer<
+      typename boost::graph_traits<Graph>::vertex_descriptor,
+      PMap, Graph,
+      CGAL::Tag_true>{
+    void operator()(PMap map, const Graph& g)
+    {
+      CGAL::helpers::init_vertex_indices(g, map);
```

Comment:
I'm still wondering if it makes sense to overwrite all the time the indices of the user map. I think it would be safer to first check that indices are not correctly initialized (by using a `std::vector<bool>(num_vertex(g))` to check for duplicate indices+each index < `num_vertex(g)`).

Reply:
@MaelRL do you agree?

Reply:
It's a problem (see issue #1918), but it seems the initialization is here only performed when a dynamic pmap has been requested (i.e. no index map in the NP); @maxGimeno?

Reply:
Actually it does seem wrong: https://github.com/CGAL/cgal/pull/4209/files#diff-aad529ba52ba39aa444683bce2b8c679R179-R195. The internal pmap should not be overwritten, whether it has already been initialized or not.

Reply:
I think it should work as follow:

- if the user pass an index map, we do nothing about it but to check that the indices are valid. 
- if the default map is used and it is writable, then we must initialize it if not already initialized
- if there is no default map and nothing is passed, use an internal dynamic one that we of course initialize. Note that If an algorithm is updating the mesh, and the index map is used after the update then we cannot use it safely.

Reply:
Currently: 
- If there is a map in the np, nothing is done to it.
- Else if there is a writable map in the mesh, we initialize it
- else we create a dynamic pmap and initialize it.

So we never overwrite the indices of the user map, only the ones of the default internal maps.

Reply:
> Actually it does seem wrong: https://github.com/CGAL/cgal/pull/4209/files#diff-aad529ba52ba39aa444683bce2b8c679R179-R195. The internal pmap should not be overwritten, whether it has already been initialized or not.

Why is that ?

Reply:
Because you can imagine a situation where a user has an existing, initialized ID map but forgot to pass it as NP. You then do not want to mess up with this ID map.

Reply:
Then you won't ever be able to use this mechanism with a Polyhedron if it has an internal id map, like in the demo.

Reply:
Let's close the loop here and implement what I suggest (safest option IMO). You have all what I suggested but when the is-initialized test for writable internal index map (I already gave the way to do it using a bitset in this thread).

Reply:
My first remark (https://github.com/CGAL/cgal/pull/4209#discussion_r357528453) is wrongly worded : I only meant "shouldn't overwrite the internal ID map if it's already been initialized" (which is @sloriot's second point). Obviously, if it's not initialized, you can use it.

Reply:
@maxGimeno I don't think the is_intialized has been implemented